src/components/ArticlesList.jsx

```diff
@@ -1,48 +1,85 @@
-import { useEffect, useState } from "react"
-import { getArticles } from "../../API"
-import { Card, Container, Row, Col, Spinner } from 'react-bootstrap';
+import { useEffect, useState } from "react";
+import { getArticleTopics, getArticles } from "../../API";
+import { Card, Container, Row, Col, Spinner, Dropdown, DropdownButton } from 'react-bootstrap';
 import { Link } from "react-router-dom";
-import '../App.css'
+import '../App.css';
 
-const ArticlesList = () => {
 
-    const [articles, setArticles] = useState([])
-    const [isLoading, setIsLoading] = useState(true)
+const ArticlesList = () => {
+    const [articles, setArticles] = useState([]);
+    const [filteredArticles, setFilteredArticles] = useState([]);
+    const [isLoading, setIsLoading] = useState(true);
+    const [selectedTopic, setSelectedTopic] = useState('');
+    const [topics, setTopics] = useState([]);
 
 
     useEffect(() => {
-        setIsLoading(true)
-        getArticles()
-            .then(({ articles }) => {
-                setArticles(articles)
-                setIsLoading(false)
+        setIsLoading(true);
+
+        Promise.all([getArticles(), getArticleTopics()])
+            .then(([articlesResponse, topicsResponse]) => {
+                setArticles(articlesResponse.articles);
+                setFilteredArticles(articlesResponse.articles); 
+                setTopics(topicsResponse.topics);
+                setIsLoading(false);
             })
             .catch((err) => {
-                console.log(err)
-            })
-    }, [])
+                console.log(err);
+                setIsLoading(false);
+            });
+    }, []);
+
+
+    useEffect(() => {
+        if (selectedTopic) {
+            const filtered = articles.filter(article => article.topic === selectedTopic);
```

Comment:
You should have an API endpoint query to do this for you, you shouldn't need the filter :) 

(if this isn't in your backend, let me know 👍)